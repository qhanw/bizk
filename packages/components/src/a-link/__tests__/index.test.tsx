import { describe, it, expect, afterEach } from 'vitest';
import { render, screen, cleanup } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';

import ALink from '../index';

describe('suite ALink', () => {
  // Default on import: runs it after each test.
  afterEach(() => {
    cleanup();
  });

  it('renders base', () => {
    const { container } = render(
      <ALink href="https://bizk.qhan.wang/">Click me!</ALink>,
    );
    expect(container).toMatchSnapshot();
  });
  it('renders click', () => {
    const { container } = render(
      <ALink onClick={() => console.log('Click!')}>Click me!</ALink>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders target', () => {
    const { container } = render(
      <ALink target="_blank" href="https://bizk.qhan.wang/">
        Click me!
      </ALink>,
    );
    expect(container).toMatchSnapshot();
  });
  it('renders to', () => {
    const { container } = render(
      <BrowserRouter>
        <ALink target="_blank" to={{ pathname: '/blank' }}>
          Click me!
        </ALink>
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders disabled ALink component', () => {
    const { container } = render(
      <ALink href="https://bizk.qhan.wang/" disabled>
        Click me!
      </ALink>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders Button ALink component', () => {
    const { container } = render(
      <ALink href="https://bizk.qhan.wang/" button={{ type: 'primary' }}>
        Click me!
      </ALink>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders ALink component', () => {
    render(<ALink href="https://bizk.qhan.wang/">Click me!</ALink>);

    expect(screen.getByText('Click me!')).toBeTruthy();
  });
});
