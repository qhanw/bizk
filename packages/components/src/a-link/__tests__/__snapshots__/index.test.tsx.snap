// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`suite ALink > renders Button ALink component 1`] = `
<div>
  <a
    class="ant-typography css-dev-only-do-not-override-ph9edi"
    href="https://bizk.qhan.wang/"
  >
     
    <button
      class="ant-btn css-dev-only-do-not-override-ph9edi ant-btn-primary"
      type="button"
    >
      <span>
        Click me!
      </span>
    </button>
  </a>
</div>
`;

exports[`suite ALink > renders base 1`] = `
<div>
  <a
    class="ant-typography css-dev-only-do-not-override-ph9edi"
    href="https://bizk.qhan.wang/"
  >
     
    Click me!
  </a>
</div>
`;

exports[`suite ALink > renders click 1`] = `
<div>
  <a
    class="ant-typography css-dev-only-do-not-override-ph9edi"
  >
     
    Click me!
  </a>
</div>
`;

exports[`suite ALink > renders disabled ALink component 1`] = `
<div>
  <a
    class="ant-typography ant-typography-disabled css-dev-only-do-not-override-ph9edi"
    href="https://bizk.qhan.wang/"
  >
     
    Click me!
  </a>
</div>
`;

exports[`suite ALink > renders target 1`] = `
<div>
  <a
    class="ant-typography css-dev-only-do-not-override-ph9edi"
    href="https://bizk.qhan.wang/"
    rel="noopener noreferrer"
    target="_blank"
  >
     
    Click me!
  </a>
</div>
`;

exports[`suite ALink > renders to 1`] = `
<div>
  <a
    href="/blank"
    target="_blank"
  >
    Click me!
  </a>
</div>
`;
